mne/viz/ica.py

```diff
@@ -206,11 +211,13 @@ def _plot_ica_sources_evoked(evoked, exclude, title, show):
     times = evoked.times * 1e3
 
     # plot unclassified sources
-    plt.plot(times, evoked.data.T, 'k')
+    if picks is None:
+        picks = np.arange(evoked.data.shape[0])
+    plt.plot(times, evoked.data[picks].T, 'k')
     for ii in exclude:
-        # use indexing to expose event related sources
-        plt.plot(times, evoked.data[ii].T, color='r', label='ICA %02d' % ii)
-
+        if ii in picks:
+            # use indexing to expose event related sources
+            plt.plot(times, evoked.data[ii].T, color='r', label='ICA %02d' % ii)
```

Comment:
so how it works is that `_transform_XXX` checks selects ica.ch_names that is a subset of evoked.ch_names. it selects those channels and returns that subset of channels to `_transform` and this ends the chain from instance to sources.
these sources are then fed into their respective `_plot_ica_sources_XXX`. the picks submitted to `ica.plot_sources` are the indices of the sources that are to be picked and plotted.

evoked.ch_names does not have to equal ica.ch_names, however, ica.ch_names must be a subset of evoked.ch_names.